### ISSUE TITLE
Generate throughput / recall graph for base on SIFT

### ISSUE THREAD

**Issue by breezykermo** [breezykermo/oak#42]
The graphs generated by default in [criterion](docs.rs/criterion) measure latency against "density":

![Image](https://github.com/user-attachments/assets/bef89896-d788-4d7c-a273-041dbd5f0767)

We want to generate throughput versus recall graphs, like in the ACORN paper. In all other uncertain respects we should also follow the ACORN paper. For example, I assume that the graphs they have are averaged over the query set in some way.

We also need to be careful when calling `search` to not factor in the code that generates the bitmask, which calling `dataset.search` will do. In other words, we need to time the `AcornHnswIndex.search`, which takes in a `filter_id_map`.

These graphs should also be generated using the appropriate queries rather than randomly generated vectors, i.e. `sift_query` vectors and attributes. (The search should be on the index constructed over the queries in `sift_base`.)

**Comment by breezykermo** [comment]
We need a QPS / recall graph for the SIFT query set on a single predicate, i.e. `=1`. 
- To vary recall, we vary `efSearch` and `efConstruction` (if necessary) on the index, as [shown here](https://github.com/csirianni/ACORN/blob/main/benchs/bench_hnsw.py#L71-L76).
- To measure QPS, we run the 'sift_query' set and divide the number of queries by total time.
- Let's do 3 experiments per recall data point, and 5 recall data points (according to the recall that is produced by the variation of `efSearch`).
- We aim to do this on a local machine with a single thread, as what we're interested in is the distance between this experiment and an experiment with oracle indexes.

**Comment by MithiJ** [comment]
For recall, it seems we have to tune efs until we get to the desired recall. [Here](https://github.com/TAG-Research/ACORN/blob/8cbf16d42fcc08073f8d42232dfc732afb60499c/demos/demo_sift1M.cpp#L230) is how they implement this for benchmarking ACORN. It seems they just check the result against the groundtruth and compute how many of the top 1 or 10 or 100 elements from ground truth show up in the results from ACORN. So, they output accuracy in top-k. `nq` seems to be the total number of queries. Thus, the outputted recall is a measure of how many correct nearest neighbors ACORN can return. 
Not sure how they fix recall at 0.9. They mention fixing it at 0.9 using efs and perhaps once they generate the recall-QPS curve, they pick the point at which recall hits 90% accuracy.

```
We perform a grid search for 𝑀 ∈{32,64,128}and
efc ∈{40,80,120,160,200}and choose the pair that obtains the highest QPS at 0.9 Recall for ANN
search. For TripClick, we choose 𝑀= 128,efc= 200. We generate each recall-QPS curve by varying
the search parameter efs from 10 to 800 in step sizes of 50.
```
^from the paper
